### DISCUSSION TITLE
How to add projects to the visual studio designer from a module

### DISCUSSION THREAD

**Discussion by stephanjohnson** [IntentArchitect/Support#9] [Q&A] [answered]
I want to develop a module that helps me scaffold my unit test projects. I have the following goals:

1) add unit test projects to my solution that are managed by the solution designer
2) scaffold some base classes that assist in providing data for my unit tests
3) automatically create and scaffold partial classes and test methods for domain entities, operations, services, or other models such as domain or integration events

I am unsure of how to alter the solution model from a module.

**Comment by joelsteventurner** [comment, marked as answer]
Hi @stephanjohnson 

To alter the solution model from a module, the following "How To" should have all the info you need.
https://docs.intentarchitect.com/articles/application-templates/metadata-installation/metadata-installation.html

It covers installing Metadata into designers (the example just happens to be the Visual Studio Designer), the second half of the "How To" specifically covers doing this from a Module.

We also have a Unit Testing Module named `Intent.Application.MediatR.CRUD.Tests` in our repository which you could take a look at, it might be useful or give you some ideas in this regard.

**Reply by stephanjohnson** [reply]
Thanks @joelsteventurner,

I'll check out the `Intent.Application.MediatR.CRUD.Tests` and the tutorial. Will provide feedback if I run into any issues.
